cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/EvaluationOrderGraphPass.kt

```diff
@@ -1028,6 +1029,10 @@ open class EvaluationOrderGraphPass(ctx: TranslationContext) : TranslationUnitPa
         pushToEOG(stmt)
     }
 
+    protected fun handleThrowStatement(statement: ThrowStatement) {
+        TODO("Needs implementing")
```

Comment:
```suggestion
        createEOG(statement.exception)
        createEOG(statement.cause)
        pushToEOG(statement)
```
Is this the right order of cause and exception?

Reply:
However, I'm not sure how to connect the `statement.nextEOG` edge to either a `CatchStatement` (if it's surrounded by a try-catch with a matching catch) the FunctionDeclaration or if this will be subject of this PR at all. Might also be a separate PR to not block other developments.

Reply:
In theory we might actually have all the necessary information because the EOG runs after the type resolver and we can the get the type information out of the `CatchStatement` and then match it. If there is no match, we should probably connect the EOG to the function declaration

Reply:
Ah lol we already have a (working?) implementation in handleThrowOperator! This handles the C++ implementation with a `throw` unary operator. But we have a problem here... This relies on a `TryScope` to find a matching TryStatement. But in python we do not open a scope for `try` since it does not open a new (variable) scoping. This is now the problem that we are mixing scopes for variable visibility and other stuff :(

Additionally, the implementation  is not yet handling nested try statements.